### DISCUSSION TITLE
Add Additional/Custom Relationships to Roles

### DISCUSSION THREAD

**Discussion by Base29** [spatie/laravel-permission#2137] [Q&A] [answered]
Hello ... I require to add additional relationships to roles. I am not sure how can I achieve that. At the moment I have created a custom model class `Role.php` that extends `\Spatie\Permission\Models\Role`

`
namespace App\Models;

use App\Models\EmployeeHandbook;
use App\Models\InductionChecklist;
use App\Models\InterviewPolicy;
use App\Models\LocumSession;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Role extends \Spatie\Permission\Models\Role
{
    use HasFactory;

    protected $hidden = [
        'guard_name',
        'created_at',
        'updated_at',
        'pivot',
    ];

    public function inductionChecklist()
    {
        return $this->hasMany(InductionChecklist::class);
    }

    public function interviewPolicy()
    {
        return $this->hasOne(InterviewPolicy::class);
    }

    public function hasInterviewPolicy()
    {
        return InterviewPolicy::where('role_id', $this->id)->first();
    }

    public function locumSessions()
    {
        return $this->hasMany(LocumSession::class);
    }

    public function handbooks()
    {
        return $this->belongsToMany(EmployeeHandbook::class);
    }

    public function itPolicies()
    {
        return $this->belongsToMany(ItPolicy::class);
    }
}
`

But when I try to pull `roles.itPolicies` like this 

`User::where('email', $request->email)
            ->with('profile', 'positionSummary', 'contractSummary', 'roles.itPolicies', 'practices')
            ->firstOrFail();`

I get this error message

`Call to undefined relationship [itPolicies] on model [Spatie\\Permission\\Models\\Role].`

Please can you guide me on how can I achieve this? I would really appreciate any help regarding this matter. Thanks in advance.

**Comment by erikn69** [comment, marked as answer]
Just read documentation
https://spatie.be/docs/laravel-permission/v5/advanced-usage/extending#content-extending-role-and-permission-models
